# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux/eth-naming.patch
# Copyright (C) 2006 - 2018 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Ever chaning wlan^Wwifi^Weth interface names do not only annoy users, but
also prevent reusing configuration from one machine to another.

Sanitze naming to alway be wlan* regardless of who wrote the driver.

  - Rene Rebe <rene@exactcode.de>

--- linux-3.17/drivers/net/wireless/cisco/airo.c.vanilla	2014-10-08 19:13:40.933902715 +0000
+++ linux-3.17/drivers/net/wireless/cisco/airo.c	2014-10-08 19:15:20.797903310 +0000
@@ -2685,7 +2685,7 @@
 					struct net_device *ethdev)
 {
 	int err;
-	struct net_device *dev = alloc_netdev(0, "wifi%d", NET_NAME_UNKNOWN,
+	struct net_device *dev = alloc_netdev(0, "wlan%d", NET_NAME_UNKNOWN,
 					      wifi_setup);
 	if (!dev)
 		return NULL;
--- linux-4.12/drivers/net/wireless/marvell/mwifiex/main.c.vanilla	2017-07-06 14:02:56.949536458 +0200
+++ linux-4.12/drivers/net/wireless/marvell/mwifiex/main.c	2017-07-06 14:03:08.189536198 +0200
@@ -594,7 +594,7 @@
 
 	rtnl_lock();
 	/* Create station interface by default */
-	wdev = mwifiex_add_virtual_intf(adapter->wiphy, "mlan%d", NET_NAME_ENUM,
+	wdev = mwifiex_add_virtual_intf(adapter->wiphy, "wlan%d", NET_NAME_ENUM,
 					NL80211_IFTYPE_STATION, NULL);
 	if (IS_ERR(wdev)) {
 		mwifiex_dbg(adapter, ERROR,
--- linux-4.16/drivers/net/wireless/cisco/airo.c.vanilla	2018-05-24 14:00:33.388748630 +0000
+++ linux-4.16/drivers/net/wireless/cisco/airo.c	2018-05-24 14:01:30.184749916 +0000
@@ -2846,7 +2846,7 @@
 		set_bit(FLAG_FLASHING, &ai->flags);
 	}
 
-	strcpy(dev->name, "eth%d");
+	strcpy(dev->name, "wlan%d");
 	rc = register_netdev(dev);
 	if (rc) {
 		airo_print_err(dev->name, "Couldn't register_netdev");
